katana/native/file_manager/file_manager_utils/ftp_utils.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python3
+import os
+import ftplib
+from ftplib import FTP
+from katana.native.file_manager.views import getpath
+
+def ftpfile(host, port, username, passwd, destdir, files_path, files_name, all_files_path, request):
+    """
+    The main FTP function which takes in input details and FTPs the files.
+    A FTP session is created.
+
+    :param host: User Input (IP address)
+    :param port: User Input (Default 21)
+    :param username: Username of the Target machine
+    :param passwd: Password of the target machine
+    :param destdir: Location from the root directory in the Target Machine where the files should be FTPed into
+    :param files_path: Full path of the top level files/directories selected
+    :param files_name: Names of the top level files/directories selected
+    :param all_files_path: List of All the files/directories selected (due to down cascading)
+    :return: Error message or list of files FTPed with status.
+    """
+    result = []
+    os.chdir(getpath(request))
+    try:
+        ftp = FTP('')
+        ftp.connect(host, port)
+        ftp.login(username, passwd)
+        try:
+            if(change_destination(ftp, destdir)):
+                for path, name in zip(files_path, files_name):
+                    if os.path.isfile(path):
+                        result.append(transfer_file(ftp, path, name))
+                    elif os.path.isdir(path):
+                        result.append(name + ': Is a directory')
+                        result.append(isDir(ftp, path, name, files_path, files_name, all_files_path))
+                    else:
+                        result.append(name + ': Is not a file or directory')
+            else:
+                result.append(destdir + " :Could Not Change to Destination Directory Successfully")
+                return result
+        except ftplib.all_errors as e:
+            result.append(str(e))
+        ftp.quit()
+    except ftplib.all_errors as e:
+        result.append(str(e))
+    return result
+
+
+def change_destination(ftp, destdir):
+    """
+    If the destiantion directory is different than the root in the target machine, then this function traverses into
+    the destination directory in the target machine.
+
+    :param ftp: FTP object
+    :param destdir: Destination directory from the root directory where the files are to be FTPed.
+    :return: Boolean.
+    """
+    try:
+        if destdir != '':
+            ftp.cwd(destdir)
+    except:
```

Comment:
Please catch a specific exception and print out a an error accordingly. Please also a catch general exceptions and print those exceptions out.

Reply:
Done.